packages/notifications/src/elements/content/Close.tsx

```diff
@@ -6,18 +6,11 @@
  */
 
 import React, { HTMLAttributes } from 'react';
-import PropTypes from 'prop-types';
-import { StyledClose, IStyledCloseProps } from '../../styled';
+import { StyledClose } from '../../styled';
 
 /**
  * Used to close a Notification. Supports all `<button>` props
  */
-export const Close = React.forwardRef<
-  HTMLButtonElement,
-  IStyledCloseProps & HTMLAttributes<HTMLButtonElement>
->((props, ref) => <StyledClose ref={ref} {...props} />);
-
-Close.propTypes = {
-  isFocused: PropTypes.bool,
-  isHovered: PropTypes.bool
-};
+export const Close = React.forwardRef<HTMLButtonElement, HTMLAttributes<HTMLButtonElement>>(
+  (props, ref) => <StyledClose ref={ref} {...props} />
+);
```

Comment:
> We should be using context to get the right type information to the Close component rather than expecting a consumer to match a Close type with its containing notification element.

@jzempel, I'm trying to understand the problems around `Close` so I can make the appropriate changes. How are consumers matching a `Close` type with its containing notification element? Does "consumers" refer to end consumers that use the `react-notifications` package like this? When you refer to "type information", is that one of "success", "error", "warning", or "info"?

```jsx
<Alert type="success">
  <Title>Success Alert</Title>
  Veggies es bonus vobis, proinde vos postulo essum magis kohlrabi welsh onion daikon.
  <Close onClick={() => alert('closing alert')} aria-label="Close Alert" />
</Alert>
```

Reply:
I'm not exactly sure this helps, but this is the way I think about the problem. The table shows that I can gather all the context needed at the Alert/Notification/Well element level to render whatever the correct treatment should be for each of the Base/Title/Close children. This means the `Styled` components don't need to establish a bunch of CSS cross-references. They simply render whatever hue they are given by the Title/Close wrapping elements.

| Element | type | StyledBase hue | StyledTitle hue | StyledClose hue |
| -------- | ----- | --------------- | --------------- | ---------------- |
| Alert | Success | green | green | green |
| Alert | Error | red | red | red |
| Alert | Warning | yellow | yellow | yellow |
| Alert | Info | grey | grey | grey |
| Notification | Success | `undefined` | green | grey |
| Notification | Error | `undefined` | red | grey |
| Notification | Warning | `undefined` | yellow | grey |
| Notification | Info | `undefined` | grey | grey |
| Well | `isRecessed === false` | `undefined` | grey | grey |
| Well | `isRecessed === true` | grey | grey | grey |

In other words, you shouldn't need to have `${StyledClose} { ... }` CSS scattered all over the place. It can all just live within the `StyledClose` component.

Reply:
Important that we would want to store the hue names (i.e. `successHue`, `errorHue`, `warningHue`, `neutralHue`) in context rather than the colors.